src/AccessibilityInsights.SharedUx/Controls/ScannerResultControl.xaml.cs

```diff
@@ -275,65 +276,46 @@ private void lvDetails_SizeChanged(object sender, SizeChangedEventArgs e)
                 Logger.PublishTelemetryEvent(TelemetryAction.Scan_File_Bug, new Dictionary<TelemetryProperty, string>() {
                     { TelemetryProperty.By, FileBugRequestSource.HowtoFix.ToString() },
-                    { TelemetryProperty.IsAlreadyLoggedIn, BugReporter.IsConnected.ToString(CultureInfo.InvariantCulture) }
+                    { TelemetryProperty.IsAlreadyLoggedIn, IssueReporter.IsConnected.ToString(CultureInfo.InvariantCulture) }
                 });
 
-                // TODO: figuring out whether a team project has been chosen should not require
-                //  looking at the most recent connection, this should be broken out
-                if (BugReporter.IsConnected && Configuration.SavedConnection?.IsPopulated == true)
+                if (IssueReporter.IsConnected)
                 {
-                    Action<int> updateZoom = (int x) => Configuration.ZoomLevel = x;
-                    (int? bugId, string newBugId) = FileBugAction.FileNewBug(vm.GetBugInformation(), Configuration.SavedConnection, Configuration.AlwaysOnTop, Configuration.ZoomLevel, updateZoom);
-
-                    vm.BugId = bugId;
-
-                    // Check whether bug was filed once dialog closed & process accordingly
-                    if (vm.BugId.HasValue)
+                    IssueInformation issueInformation = vm.GetIssueInformation();
+                    FileIssueAction.AttachIssueData(issueInformation, this.EcId, vm.Element.BoundingRectangle, vm.Element.UniqueId);
+                    IIssueResult issueResult = FileIssueAction.FileIssueAsync(issueInformation);
+                    if (issueResult != null)
                     {
-                        vm.LoadingVisibility = Visibility.Visible;
-                        try
-                        {
-                            var success = await FileBugAction.AttachBugData(this.EcId, vm.Element.BoundingRectangle, 
-                                vm.Element.UniqueId, newBugId, vm.BugId.Value).ConfigureAwait(false);
-                            if (!success)
-                            {
-                                MessageDialog.Show(Properties.Resources.ScannerResultControl_btnFileBug_Click_There_was_an_error_identifying_the_created_bug_This_may_occur_if_the_ID_used_to_create_the_bug_is_removed_from_its_AzureDevOps_description_Attachments_have_not_been_uploaded);
-                                vm.BugId = null;
-                            }
-                            vm.LoadingVisibility = Visibility.Collapsed;
-                        }
-                        catch (Exception)
-                        {
-                            vm.LoadingVisibility = Visibility.Collapsed;
-                        }
+                        vm.IssueDisplayText = issueResult.DisplayText;
+                        vm.IssueLink = issueResult.IssueLink;
                     }
+                    File.Delete(issueInformation.TestFileName);
```

Comment:
This cleanup gets skipped if an exception gets thrown. Should it be in a finally block to ensure proper cleanup?

Reply:
Nope, if you mean exception in FileIssueAction.FileIssueAsync, it's already caught in the same function so still will hit this line

Reply:
Not what I meant. Suppose that an exception occurs inside FileIssueAction.AttachIssueData, but after the temp file has been created. The Exception will cause the File.Delete to get skipped, so the temp file is orphaned. If lines 306-313 were wrapped in a try block, and the File.Delete put into a finally block, then you always clean up successfully. One note: File.Delete throws if the file doesn't exist, so the finally block would be `if(File.Exists()) File.Delete;`